com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/MaxRectsLayoutStrategy.java

```diff
@@ -174,7 +182,11 @@ private Page packPage(ArrayList<RectNode> inputRects) {
     /** @param fully If true, the only results that pack all rects will be considered. If false, all results are considered, not all
      *           rects may be packed.
      **/
-    private Page packAtSize(boolean fully, int width, int height, ArrayList<RectNode> inputRects) {
+    private Page packAtSize(boolean fully, int width, int height, ArrayList<RectNode> inputRects, RectStats rectStats) {
+        if (!canFitInPage(width, height, rectStats, fully)) {
+            return null;
+        }
```

Comment:
We know the area of all images. If they can't fit the current atlas size, just skip the heuristics for that size.